### PR TITLE
EOS-16025: Conf Store: Cs indexing

### PR DESCRIPTION
**Pull request by suryakumar1024**
Added indexing support for confStore for below mentioned interfaces

confStore.get()
confStore.set()
confStore.delete()

### CODE REVIEW

File: py-utils/src/utils/kv_store/kv_store.py [hunk 1]
```diff
@@ -76,36 +77,109 @@ def __init__(self, data: dict, delim='>'):
...
-            data[k[0]] = val
-            return
+
+        # Check if key has index, if so identify index
+        index = None
+        ki = re.split(r'\W+', k[0])
+        if len(ki) > 1:
+            if not (ki[0] and ki[0].strip()):
+                raise KvStoreError(errno.EINVAL, "Invalid key name %s", ki[0])
+            # if ki[1] is not numeric then raise error
```
Comment (ujjwalpl) [inline py-utils/src/utils/kv_store/kv_store.py:87]: Please remove this comment, it does not add any value.

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/kv_store/kv_store.py [hunk 2]
```diff
@@ -76,36 +77,109 @@ def __init__(self, data: dict, delim='>'):
...
         k = key.split(self._delim, 1)
-        if len(k) == 1:
-            data[k[0]] = val
-            return
+
+        # Check if key has index, if so identify index
+        index = None
+        ki = re.split(r'\W+', k[0])
+        if len(ki) > 1:
+            if not (ki[0] and ki[0].strip()):
```
Comment (ujjwalpl) [inline py-utils/src/utils/kv_store/kv_store.py:85]: `if len(ki[0].strip()) == 0:`

File: py-utils/src/utils/kv_store/kv_store.py (hunk 1 above)
Comment (suryakumar1024) [reply py-utils/src/utils/kv_store/kv_store.py]: updated in current patch

**Review by sachinpunadikar — APPROVED** [review]
+1

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/kv_store/kv_store.py (hunk 2 above)
Comment (suryakumar1024) [reply py-utils/src/utils/kv_store/kv_store.py]: Updated in current patch.

File: py-utils/src/utils/kv_store/kv_store.py [hunk 3]
```diff
@@ -76,36 +77,108 @@ def __init__(self, data: dict, delim='>'):
...
         return self._set(key, val, self._data)
 
     def _get(self, key: str, data: dict) -> str:
         k = key.split(self._delim, 1)
+
+        # Check if key has index, if so identify index
+        index = None
+        ki = re.split(r'\W+', k[0])
+        if len(ki) > 1:
+            if not (ki[0] and ki[0].strip()):
```
Comment (ujjwalpl) [inline py-utils/src/utils/kv_store/kv_store.py]: if len(ki[0].strip()) == 0:

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

**Pull request merged**